tests/system/python/api/test_statistics.py

```diff
@@ -83,22 +83,25 @@ def test_default_statistics_history(self, fledge_url):
         assert {} == jdoc['statistics'][0]
 
     def test_statistics_history_with_stats_collector_schedule(self, fledge_url, wait_time, retries):
-        # wait for sometime for stats collector schedule to start
-        time.sleep(wait_time * retries)
-
         conn = http.client.HTTPConnection(fledge_url)
-        conn.request("GET", '/fledge/statistics/history')
-        r = conn.getresponse()
-        assert 200 == r.status
-        r = r.read().decode()
-        jdoc = json.loads(r)
-        assert len(jdoc), "No data found"
-        assert 15 == jdoc['interval']
-        assert 1 == len(jdoc['statistics'])
-        statistics = jdoc['statistics'][0]
-        assert Counter(STATS_HISTORY_KEYS) == Counter(statistics.keys())
-        del statistics['history_ts']
-        assert Counter([0, 0, 0, 0, 0, 0]) == Counter(statistics.values())
+        for i in range (0,retries):
+            # wait for sometime for stats collector schedule to start
+            time.sleep(wait_time)
+
+            conn.request("GET", '/fledge/statistics/history')
+            r = conn.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert len(jdoc), "No data found"
+            assert 15 == jdoc['interval']
+            assert 1 == len(jdoc['statistics'])
+            statistics = jdoc['statistics'][0]
+            if i <= retries and Counter(STATS_HISTORY_KEYS) != Counter(statistics.keys()): continue
+            assert Counter(STATS_HISTORY_KEYS) == Counter(statistics.keys())
+            del statistics['history_ts']
+            assert Counter([0, 0, 0, 0, 0, 0]) == Counter(statistics.values())
```

Comment:
will not assert make tests fail if not as expected in initial iterations? may be we can check/wait for stats history task?

Reply:
In the first iteration, we have a sleep at the start and then also checked with `if i <= retries and Counter(STATS_HISTORY_KEYS) != Counter(statistics.keys()): continue`; if we do not get anything it will continue the loop